source/btle/btle_discovery.cpp

```diff
@@ -44,7 +47,7 @@ void bleGattcEventHandler(const ble_evt_t *p_ble_evt)
 
                 case BLE_GATT_STATUS_ATTERR_ATTRIBUTE_NOT_FOUND:
                 default:
-                    sdSingleton.terminateCharacteristicDiscovery();
+                    sdSingleton.terminateCharacteristicDiscovery(BLE_ERROR_NONE);
```

Comment:
you're combining several different changes into a single pull request. This makes it difficult for the reviewer to manage complexity. Please try to keep pull requests focussed on a single topic at a time. 

If you wanted to improve the API for `terminateCharacteristicDiscovery`, you could have done that separately as an independent pull request.